x/bridge/observer/bitcoin/bitcoin.go

```diff
@@ -16,170 +17,170 @@ import (
 	vaultAddr string,
-	initialHeight uint64,
 	observeSleepPeriod time.Duration,
-) (Observer, error) {
+	lastObservedHeight uint64,
+) (*ChainClient, error) {
 	if len(vaultAddr) == 0 {
-		return Observer{}, errorsmod.Wrapf(ErrInvalidCfg, "Invalid vaultAddr")
+		return nil, errorsmod.Wrapf(ErrInvalidCfg, "Invalid BTC vault address")
 	}
 
-	return Observer{
-		logger:             logger.With("module", ModuleNameObserver),
+	return &ChainClient{
+		logger:             logger.With("module", ModuleName),
 		btcRpc:             btcRpc,
 		vaultAddr:          vaultAddr,
-		currentHeight:      initialHeight,
-		globalTxInChan:     make(chan TxIn),
 		stopChan:           make(chan struct{}),
+		outboundChan:       make(chan observer.Transfer),
 		observeSleepPeriod: observeSleepPeriod,
+		lastObservedHeight: lastObservedHeight,
 	}, nil
 }
 
-// Start starts observing BTC blocks for incoming Txs to the given address
-func (o *Observer) Start() {
-	go o.observeBlocks()
+// Start starts observing Bitcoin blocks for outbound transfers
+func (b *ChainClient) Start(context.Context) error {
+	go b.observeBlocks()
+
+	return nil
+}
+
+// Stop stops observing Bitcoin blocks and shutdowns RPC client
+func (b *ChainClient) Stop(context.Context) error {
+	close(b.stopChan)
+	b.btcRpc.Shutdown()
+	b.btcRpc.WaitForShutdown()
+	return nil
+}
+
+// ListenOutboundTransfer returns receive-only channel with outbound transfer items
+func (b *ChainClient) ListenOutboundTransfer() <-chan observer.Transfer {
+	return b.outboundChan
+}
+
+// SignalInboundTransfer sends `InboundTransfer` to Bitcoin
+func (b *ChainClient) SignalInboundTransfer(ctx context.Context, in observer.Transfer) error {
+	return fmt.Errorf("Not implemented")
 }
 
-// Stop stops observation loop and RPC client
-func (o *Observer) Stop() {
-	close(o.stopChan)
-	o.btcRpc.Shutdown()
-	o.btcRpc.WaitForShutdown()
+// Returns current height of the Bitcoin chain
+func (b *ChainClient) Height() (uint64, error) {
+	height, err := b.btcRpc.GetBlockCount()
+	if err != nil {
+		return 0, errorsmod.Wrapf(ErrRpcClient, "Failed to get current height %s", err.Error())
+	}
+	return uint64(height), nil
 }
 
-// TxIns returns receive-only part of observed Txs channel
-func (o *Observer) TxIns() <-chan TxIn {
-	return o.globalTxInChan
+// Returns number of required tx confirmations
+func (b *ChainClient) ConfirmationsRequired() (uint64, error) {
+	return 0, nil
 }
 
-func (o *Observer) CurrentHeight() uint64 {
-	return o.currentHeight
+func (b *ChainClient) observeBlocks() {
+	defer close(b.outboundChan)
+
+	for {
+		select {
+		case <-b.stopChan:
+			return
+		default:
+			err := b.fetchNewBlock()
+			if err != nil {
+				// Do not log error if block with this height doesn't exist yet
+				if !errors.Is(err, ErrBlockUnavailable) {
+					b.logger.Error(fmt.Sprintf(
+						"Failed to fetch block %d: %s",
+						b.lastObservedHeight+1,
+						err.Error(),
+					))
+				}
+				time.Sleep(b.observeSleepPeriod)
```

Comment:
Do you know if there is a way to subscribe to new block events on the Bitcoin network? I like the sleep period for resiliency but am curious if we can have a new block instantly trigger a observation so there is little delay in an already slow process